db/schema_tables.cc

```diff
@@ -805,31 +812,35 @@ future<mutation> query_partition_mutation(service::storage_proxy& proxy,
     partition_key pkey)
 {
     auto dk = dht::decorate_key(*s, pkey);
-    return do_with(dht::partition_range::make_singular(dk), [&proxy, dk, s = std::move(s), cmd = std::move(cmd)] (auto& range) {
-        return proxy.query_mutations_locally(s, std::move(cmd), range, db::no_timeout, tracing::trace_state_ptr{})
-                .then([dk = std::move(dk), s](rpc::tuple<foreign_ptr<lw_shared_ptr<reconcilable_result>>, cache_temperature> res_hit_rate) {
+    auto range = dht::partition_range::make_singular(dk);
+    {
+        auto res_hit_rate = co_await proxy.query_mutations_locally(s, std::move(cmd), range, db::no_timeout, tracing::trace_state_ptr{});
+        {
                     auto&& [res, hit_rate] = res_hit_rate;
                     auto&& partitions = res->partitions();
                     if (partitions.size() == 0) {
-                        return mutation(s, std::move(dk));
+                        co_return mutation(s, std::move(dk));
                     } else if (partitions.size() == 1) {
-                        return partitions[0].mut().unfreeze(s);
+                        co_return partitions[0].mut().unfreeze(s);
                     } else {
                         throw std::invalid_argument("Results must have at most one partition");
                     }
-                });
-    });
+        }
+    }
 }
 
 future<schema_result_value_type>
-read_schema_partition_for_keyspace(distributed<service::storage_proxy>& proxy, const sstring& schema_table_name, const sstring& keyspace_name)
+read_schema_partition_for_keyspace(distributed<service::storage_proxy>& proxy, const sstring& schema_table_name_, const sstring& keyspace_name_)
```

Comment:
Why not just change to call-by-value? I know I too use this pattern occasionally. I am just extra provoked by the trailing '_':s, which are to me as a Kirk is to a Kahn.
But also, since this is a function returning a future, is it not the calling frames responsibility to keep the strings ref:ed here alive?

Reply:
I plan to change it to call by value, but didn't for this series so rebases had only one file to compile. I think I mentioned it in the changelog. I'll also follow up with a reindent.